File: packages/clawde-sdk/src/context-resolver.ts
import { readFileSync, existsSync } from 'fs'
import { join } from 'path'
import { homedir } from 'os'
import { ResolvedContext } from './types'

export async function resolveContext(): Promise<ResolvedContext> {
  const configPath = join(homedir(), '.intutic', 'config.json')

  // 1. Primary: Read from sync-daemon's config (written every 30s)
  if (existsSync(configPath)) {
    try {
      const config = JSON.parse(readFileSync(configPath, 'utf-8'))
      return {
        gitBranch: config.gitBranch,
        jiraTicket: config.jiraTicket,
        pagerdutyIncident: config.pagerdutyIncident,
        ciPipeline: config.ciPipeline,
        workingDirectory: config.workingDirectory || process.cwd(),
        workspaceId: config.workspaceId,
        sessionId: config.sessionId,
      }
    } catch {
      // JSON parse failed or config locked, fallback to env vars
    }
  }

  // 2. Fallback: Environment variables (if sync-daemon not running)
  return {
    workspaceId: process.env.INTUTIC_WORKSPACE_ID,
    sessionId: process.env.INTUTIC_SESSION_ID,
    gitBranch: process.env.GIT_BRANCH,
    ciPipeline: process.env.GITHUB_RUN_ID
      || process.env.BUILDKITE_BUILD_ID
      || process.env.CIRCLE_BUILD_NUM,
    pagerdutyIncident: process.env.PD_INCIDENT_ID,
    workingDirectory: process.cwd(),
  }
}

File: packages/intutic-clawde/intutic_clawde/context_resolver.py
import os
import json
from pathlib import Path
from typing import Dict, Any

def resolve_context() -> Dict[str, Any]:
    config_path = Path.home() / ".intutic" / "config.json"
    
    # 1. Primary: Read from sync-daemon config
    if config_path.exists():
        try:
            with open(config_path, "r", encoding="utf-8") as f:
                config = json.load(f)
                return {
                    "gitBranch": config.get("gitBranch"),
                    "jiraTicket": config.get("jiraTicket"),
                    "pagerdutyIncident": config.get("pagerdutyIncident"),
                    "ciPipeline": config.get("ciPipeline"),
                    "workingDirectory": config.get("workingDirectory", os.getcwd()),
                    "workspaceId": config.get("workspaceId"),
                    "sessionId": config.get("sessionId"),
                }
        except Exception:
            pass
            
    # 2. Fallback: Environment variables
    return {
        "workspaceId": os.environ.get("INTUTIC_WORKSPACE_ID"),
        "sessionId": os.environ.get("INTUTIC_SESSION_ID"),
        "gitBranch": os.environ.get("GIT_BRANCH"),
        "ciPipeline": os.environ.get("GITHUB_RUN_ID") 
            or os.environ.get("BUILDKITE_BUILD_ID") 
            or os.environ.get("CIRCLE_BUILD_NUM"),
        "pagerdutyIncident": os.environ.get("PD_INCIDENT_ID"),
        "workingDirectory": os.getcwd(),
    }
